implementations/elixir/ockam/ockam/lib/ockam/credential/authenticator/direct/client.ex

```diff
@@ -0,0 +1,121 @@
+defmodule Ockam.Credential.Authenticator.Direct.Client.AddMemberRequest do
+  @moduledoc """
+  Request data structure for Credential Authority authenticator API add_member
+  """
+  use TypedStruct
+
+  typedstruct do
+    plugin(Ockam.TypedCBOR.Plugin)
+    field(:identity_id, String.t(), minicbor: [key: 1])
+    field(:attributes, %{String.t() => String.t()}, minicbor: [key: 2])
+  end
+end
+
+defmodule Ockam.Credential.Authenticator.Direct.Client.AttributesEntry do
+  @moduledoc """
+  Data structure representing attribute entry for Credential Authority authenticator API list_members
+  """
+  use TypedStruct
+
+  typedstruct do
+    plugin(Ockam.TypedCBOR.Plugin)
+    ## Rust encodes attribute values as a list of bytes
+    field(:attributes, %{String.t() => {:list, :integer}}, minicbor: [key: 1])
+    field(:added_at, integer(), minicbor: [key: 2])
+    field(:expires, integer() | nil, minicbor: [key: 3])
+    field(:attested_by, String.t() | nil, minicbor: [key: 4])
+  end
+end
+
+defmodule Ockam.Credential.Authenticator.Direct.Client.ListMembersResponse do
+  @moduledoc """
+  Response data structure for Credential Authority authenticator API list_members
+
+  This API returns a map of identity identifiers to attribute entries
+  """
+
+  alias Ockam.Credential.Authenticator.Direct.Client.AttributesEntry
+
+  @schema {:map, :string, AttributesEntry.minicbor_schema()}
+
+  def decode_strict(data) do
+    with {:ok, decoded} <- Ockam.TypedCBOR.decode_strict(@schema, data) do
+      ## Convert attribute values to binaries
+
+      Enum.map(decoded, fn {id, entry} ->
+        attributes =
+          Map.get(entry, :attributes, %{})
+          |> Enum.map(fn {key, val} -> {key, :erlang.list_to_binary(val)} end)
+          |> Map.new()
```

Comment:
This is directly equivalent:
```suggestion
          |> Map.new(fn {key, val} -> {key, :erlang.list_to_binary(val)} end)
```